### PR TITLE
Add FlxSound demo, add slight tweak to mode demo (custom FPS counter, hidden by default)

### PR DESCRIPTION
**Pull request by larsiusprime**
These are some changes to aid with testing the openfl-next transition. 

1. FlxSound demo that tests the major features of the FlxSound api (music, sfx, volume, looping, etc).
2. Custom FPS counter for Mode, hidden by default. Useful for sanity-checking performance diffs.

### CODE REVIEW

File: Platformers/Mode/source/PlayState.hx [hunk 1]
```diff
@@ -515,4 +521,69 @@ class PlayState extends FlxState
 		
 		return _map;
 	}
+	
+	#if SHOW_FPS
```
Comment (Gama11) [inline Platformers/Mode/source/PlayState.hx]: Couldn't we use `openfl.display.FPS` instead? The FlxNape demo does that.

Comment (larsiusprime) [inline Platformers/Mode/source/PlayState.hx]: Sure could! I didn't know such a facility existed.

File: Flixel Features/FlxSound/source/MenuState.hx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	
+	/**
+	 * Function that is called up when to state is created to set it up. 
+	 */
+	override public function create():Void
+	{
+		_xml_id = "menu";
+		super.create();
+		
+		var butt_music:FlxUIButton = cast _ui.getAsset("butt_music");
```
Comment (Gama11) [inline Flixel Features/FlxSound/source/MenuState.hx]: `butt_`, huh? Interesting abbreviation for button. :D

Comment (larsiusprime) [inline Flixel Features/FlxSound/source/MenuState.hx]: Oh I've always done that, guess I have the mind of a teenager :P I can change to btn_ if you insist.

Comment (Gama11) [inline Flixel Features/FlxSound/source/MenuState.hx]: I don't really care, just seemed a little odd.

**Comment by larsiusprime** [comment]
@Gama11: Is that better?

**Comment by Gama11** [comment]
I'm getting a crash on neko when changing the sfx or music volume before playing sfx or music.

**Comment by Gama11** [comment]
The FlxSound demo doesn't seem to be working on flash? I think you specifically need to check for `AssetType.SOUND` and `AssetType.MUSIC` in `Assets.exists()`.

**Comment by Gama11** [comment]
Would be nice if the Sound demo eventually tested stuff like `pitch` (not supported in next yet btw I think), `pan` and `proximity` as well, but I guess that's not necessary for now.

**Comment by larsiusprime** [comment]
Yeah, that'd be great! We'll add that when those features come online. Flash and neko both seem to work now.

**Comment by Gama11** [comment]
Seems like I'm unable to manually enter values in the numeric steppers input text (that is, it only accepts 1 or 0, not 0.2 for example), guess that might be a FlxInputText thing with QWERTY keyboards?

**Comment by larsiusprime** [comment]
Try adjusting the number of digits in the flixel-ui xml markup and see if that helps.

**Comment by Gama11** [comment]
Already tried that, doesn't seem to make any difference. I take it it's working fine for you?

**Comment by larsiusprime** [comment]
No, doesn't work for me either. It might be as simple as me not testing input for decimal values -- "." might not even be an allowed character! Let me see if I can patch this real quick.

**Comment by larsiusprime** [comment]
Try now:
https://github.com/HaxeFlixel/flixel-ui/commit/0b9a575aa2a15cd43b02f13c7a0959819ff9b6dd

**Comment by Gama11** [comment]
I opened a separate issue over at flixel-ui: https://github.com/HaxeFlixel/flixel-ui/issues/101

**Pull request merged**